### DISCUSSION TITLE
Can I generate a .NET Core assembly with Cecil?

### DISCUSSION THREAD

**Discussion by peter-villadsen** [jbevain/cecil#799] [Q&A]
I am using cecil as an IL code generator instead of reflection.emit, which will not currently work for .NET Core as far as I know. I use AssemblyDefinition, TypeDefinitions and MethodDefinitions in the obvious ways. However, I am having trouble consuming the resulting assemblies from other .NET Core contexts. I would like to know if writing .NET Core assemblies is supported? I am worried that the TargetRuntime enum does not contain any references to .NET Core.

BR PV

**Comment by Zastai** [comment]
A typical problem is generating core/5+ assemblies when the generating code is .NET framework (or vice versa) and using `System.Type` instances to produce type references. That will cause there to be references to the wrong standard library (`mscorlib` vs `netstandard` vs `System.Private.CoreLib`).

Try looking at your generated assembly's AssemblyRef table in ILSpy and/or running it through [ILVerify](https://github.com/dotnet/runtime/tree/main/src/coreclr/tools/ILVerify) to identify what the problem is.

If the problem is such bad type imports, you'll need to either build your code generator to match the target platform of the generated assembly, or construct `TypeReference`s yourself so that they are not affected by the target framework of the generator code.
